Model/Sync/Reset/Main.php

```diff
@@ -0,0 +1,168 @@
+     */
+    protected function getYotpoEntityName()
+    {
+        return '';
+    }
+
+    /**
+     * @param int $storeId
+     * @return void
+     * @throws \Zend_Db_Statement_Exception
+     */
+    private function deleteAllFromTables($storeId)
+    {
+        $tableResourceNames = $this->getTableResourceNames();
+        foreach ($tableResourceNames as $tableResourceName) {
+            $this->deleteAllFromTable($storeId, $tableResourceName);
+        }
+    }
+
+    /**
+     * @param int $storeId
+     * @param string $tableResourceName
+     * @return void
+     * @throws \Zend_Db_Statement_Exception
+     */
+    private function deleteAllFromTable($storeId, $tableResourceName)
+    {
+        $tableName = $this->resourceConnection->getTableName($tableResourceName);
+        $totalCount = $this->getTotalCount($tableName, $storeId);
+        if (!$totalCount) {
+            return;
+        }
+        $connection  = $this->resourceConnection->getConnection();
+        while ($totalCount > 0) {
+            $totalCount -= self::DELETE_LIMIT;
+            $select = $connection
+                ->select()
+                ->from($tableName)
+                ->where('store_id = \''.$storeId.'\'')
+                ->limit(self::DELETE_LIMIT);
+
+            $query = $connection->deleteFromSelect($select, new \Zend_Db_Expr(''));
+            $connection->query($query);
+        }
+    }
+
+    /**
+     * @param string $tableName
+     * @param int $storeId
+     * @return int
+     * @throws \Zend_Db_Statement_Exception
+     */
+    private function getTotalCount($tableName, $storeId)
+    {
+        $connection  = $this->resourceConnection->getConnection();
+        $query = $connection->select()
+            ->from($tableName)
+            ->where('store_id = ?', $storeId);
+        return $connection->query($query)->rowCount();
+    }
+
+    /**
+     * @return void
+     */
+    private function deleteRunningCronSchedules()
+    {
+        $jobCodes = $this->getCronJobCodes();
+        if (!$jobCodes) {
+            return;
+        }
+        $connection  = $this->resourceConnection->getConnection();
+        $tableName = $this->resourceConnection->getTableName('cron_schedule');
+        $select = $connection
+            ->select()
+            ->from($tableName)
+            ->where('job_code IN (?)', $jobCodes)
+            ->where('status=\'running\'');
+        $query = $connection->deleteFromSelect($select, new \Zend_Db_Expr(''));
+        $connection->query($query);
+    }
+
+    /**
+     * @param int $storeId
+     * @param string $flag
+     * @return void
+     */
+    public function setResetInProgressConfig($storeId, $flag)
+    {
+        $yotpoEntityName = $this->getYotpoEntityName();
+        $key = 'reset_sync_in_progress_' . $yotpoEntityName;
+        $this->config->saveConfig($key, $flag, $storeId);
+        $this->cacheTypeList->cleanType(\Magento\Framework\App\Cache\Type\Config::TYPE_IDENTIFIER);
```

Comment:
Can you explain what is this for?

Reply:
This is to align with the below suggestion from Eran.

"Option 2 - adding internal config flag to control the cron status"

When reset sync is happening a configuration flag will be set to let the other cron processes know that the sync is being reset. The other cron processes would listen to this flag and skip their sync iterations. In this way we can get rid of "an already started cron process continuing with it's job even though the cron schedule is removed from the table"

Reply:
OK, so why do we need this?
```
$this->cacheTypeList->cleanType(\Magento\Framework\App\Cache\Type\Config::TYPE_IDENTIFIER);
```

Reply:
This command is used to clear the configuration cache. Otherwise when we set a config value, the other parts of the code wouldn't get the exact value that we saved in the config. Without this, it might keep reading the old value from cache or read null value